gpii/node_modules/flowManager/test/SolutionsRegistryDataSourceTests.js

```diff
@@ -0,0 +1,320 @@
+    var promise = fluid.promise();
+    var solutionsRegistry = {};
+    solutionsRegistry[platformId] = solutions;
+    that.repositorySolutionsRegistry = solutionsRegistry;
+    promise.resolve(that.repositorySolutionsRegistry);
+    return promise;
+};
+
+fluid.defaults("gpii.tests.solutionsRegistry.local.mockRepository.testCaseHolder", {
+    gradeNames: "fluid.test.testCaseHolder",
+    components: {
+        solutionsRegistryMockRepoLoad: {
+            type: "gpii.tests.solutionsRegistry.dataSource.local.mockLoadFromRepository",
+            options: {
+                path: gpii.tests.solutionsRegistry.path
+            }
+        }
+    },
+    modules: [{
+        name: "Solutions registry used by LFM - repository + file system based solutions",
+        expect: 2,
+        tests: [{
+            name: "Solutions registry LFM - solutions loaded from repository and looking for named registry",
+            sequence: [
+                { funcName: "fluid.log", args: ["Solutions Registry Mock Repo Load, getting named repository registry -- START"] },
+                {
+                    event: "{solutionsRegistryMockRepoLoad}.events.solutionsRegistryReady",
+                    listener: "jqUnit.assert",
+                    args: ["LFM solutionsRegistryMockRepoLoad ready"]
+                }, {
+                    funcName: "{solutionsRegistryMockRepoLoad}.get",
+                    args: [gpii.tests.solutionsRegistry.requestOptions]
+                }, {
+                    event: "{solutionsRegistryMockRepoLoad}.events.onRead",
+                    listener: "gpii.tests.solutionsRegistry.checkNamedRegistry",
+                    args: [
+                        "Expecting named solutions registry from repository",
+                        jqUnit.assertDeepEq,
+                        gpii.tests.solutionsRegistry.repositoryDarwinSolutions,
+                        "{arguments}.0"
+                    ]
+                },
+                { funcName: "fluid.log", args: ["Solutions Registry Mock Repo Load, getting named repository registry -- END"] },
+                { funcName: "fluid.log", args: ["Solutions Registry Mock Repo Load, getting named local registry -- START"] },
+                {
+                    funcName: "{solutionsRegistryMockRepoLoad}.get",
+                    args: [{os: gpii.tests.solutionsRegistry.win32PlatformId}]
+                }, {
+                    event: "{solutionsRegistryMockRepoLoad}.events.onRead",
+                    listener: "gpii.tests.solutionsRegistry.checkOneRegistry",
+                    args: [
+                        "Expecting named solutions registries, but from the local file system",
+                        gpii.tests.solutionsRegistry.win32PlatformId,
+                        "{arguments}.0"
+                    ]
+                },
+                { funcName: "fluid.log", args: ["Solutions Registry Mock Repo Load, getting all registries -- START"] }
```

Comment:
I believe you meant `END` for logging. ;)

Reply:
Yes.  Copy/paste is your friend ... and your worst enemy.